Code/Engine/Core/Scripting/ScriptWorldModule.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <Core/ResourceManager/ResourceHandle.h>
+#include <Core/Utils/IntervalScheduler.h>
+#include <Core/World/World.h>
+
+using ezScriptClassResourceHandle = ezTypedResourceHandle<class ezScriptClassResource>;
+
+class EZ_CORE_DLL ezScriptWorldModule : public ezWorldModule
+{
+  EZ_DECLARE_WORLD_MODULE();
+  EZ_ADD_DYNAMIC_REFLECTION(ezScriptWorldModule, ezWorldModule);
+
+public:
+  ezScriptWorldModule(ezWorld* pWorld);
+  ~ezScriptWorldModule();
+
+  virtual void Initialize() override;
+
+  void AddUpdateFunctionToSchedule(const ezAbstractFunctionProperty* pFunction, void* pInstance, ezTime updateInterval);
+  void RemoveUpdateFunctionToSchedule(const ezAbstractFunctionProperty* pFunction, void* pInstance);
+
+  using ReloadFunction = ezDelegate<void()>;
+  void AddScriptReloadFunction(ezScriptClassResourceHandle hScript, ReloadFunction function);
+  void RemoveScriptReloadFunction(ezScriptClassResourceHandle hScript, void* pInstance);
```

Comment:
Out of curiosity, what is this for exactly ?

Reply:
Typically we an update function or something like that on the component manager for resource reloading but that would mean that I have to implemented resource reloading for e.g. state machine states separatly. Registering a callback on a world module is a more generic way to handle resource reloading. This might be something we could generalize even further, not sure yet.